### PR TITLE
Fix Port Binding Issue During Testing

### PR DESCRIPTION
**Pull request by neeetman**
This PR addresses the issue where tests would fail if the designated port was already in use. 
```
[INFO ] 2023-10-25 07:18:47,185 [main]  com.oltpbenchmark.benchmarks.resourcestresser.TestResourceStresserBenchmark setUp - starting HSQLDB server for test [testGetTransactionType] on port [9099]
[Server@191774d6]: [Thread[HSQLDB Server @191774d6,5,main]]: run()/openServerSocket(): 
java.net.BindException: Address already in use
```
Changes:

- Modified the `setUp` method in `TestResourceStresserBenchmark` class.
- Added logic to try different ports using `ServerSocket` to find an available port.
- ~~The system will attempt up to `MAX_TRIES` to find an open port before throwing an exception.~~
- The the system will make a thorough effort in finding an available port.

### CODE REVIEW

**Comment by apavlo** [comment]
@neeetman What problem is this trying to solve in testing? If the port is in use, then the test environment is wrong not the code.

**Comment by neeetman** [comment]
> @neeetman What problem is this trying to solve in testing? If the port is in use, then the test environment is wrong not the code.

Each test requires applying for a new port. However, sometimes other services running on the server might already be occupying that port. It might be better to consider skipping the occupied ports instead of directly terminating the test with an error.

File: src/test/java/com/oltpbenchmark/api/AbstractTestCase.java [hunk 1]
```diff
@@ -110,7 +113,7 @@ public final void setUp() throws Exception {
         server.setAddress("localhost");
         server.setPort(port);
         server.setSilent(true);
-        server.setLogWriter(null);
+        server.setLogWriter(null);
```
Comment (bpkroth) [inline src/test/java/com/oltpbenchmark/api/AbstractTestCase.java:116]:
```suggestion
        server.setLogWriter(null);
```

File: src/test/java/com/oltpbenchmark/api/AbstractTestCase.java [hunk 2]
```diff
@@ -163,6 +166,22 @@ public final void setUp() throws Exception {
...
+            int port = portCounter.incrementAndGet();
+
+            if (port > MAX_PORT_NUMBER) {
+                throw new IOException("No available port found up to " + MAX_PORT_NUMBER);
+            }
+
+            try (ServerSocket testSocket = new ServerSocket(port)) {
+                return port;
+            } catch (BindException e) {
+                // This port is already in use. Continue to next port.
```
Comment (bpkroth) [inline src/test/java/com/oltpbenchmark/api/AbstractTestCase.java:180]: Please log a warning here before continuing.

**Review by bpkroth — APPROVED** [review]
LGTM.  Minor tweak suggested.

**Comment by bpkroth** [comment]
@neeetman, just noticed you closed this before merging.  Any particular reason?

**Comment by bpkroth** [comment]
> > @neeetman, just noticed you closed this before merging. Any particular reason?
> 
> Sorry, I previously noticed that 'Some checks were not successful,' so I closed it.

Ah, for future, just ping back about that.  Happy to help look into those.

That one looks spurious.  Seems to have failed in the nightly run a few times as well, so probably needs separate investigation.  I'll file an issue on that one and follow up separately.

**Pull request merged**